### PR TITLE
fix: cel type provider should return a type type

### PR DESCRIPTION
**Pull request by eddycharly**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This PR fixes cel type provider `FindStructType` by returning a type type.
From cel comment on the interface:
```
    // FindStructType returns the Type give a qualified type name.
    //
    // For historical reasons, only struct types are expected to be returned through this
    // method, and the type values are expected to be wrapped in a TypeType instance using
    // TypeTypeWithParam(<structType>).
    //
    // Returns false if not found.
    FindStructType(structType string) (*Type, bool)
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jpbetz** [comment]
Thanks!  Any way we could get a test added for this as well?

File: staging/src/k8s.io/apiserver/pkg/cel/types.go [hunk 1]
```diff
@@ -429,7 +429,7 @@ func (rt *DeclTypeProvider) FindStructType(typeName string) (*types.Type, bool)
 	declType, found := rt.findDeclType(typeName)
 	if found {
 		expT := declType.CelType()
-		return expT, found
+		return types.NewTypeTypeWithParam(expT), found
```
Comment (cici37) [inline staging/src/k8s.io/apiserver/pkg/cel/types.go:432]: The returned celType is actually a typeType defined in https://github.com/google/cel-go/blob/7c131686c5fc852e9972934effe40e7cc6be5331/cel/decls.go#L131-L132
I don't understand why we need a duplicated conversion here.

Comment (eddycharly) [reply staging/src/k8s.io/apiserver/pkg/cel/types.go:432]: @cici37 it's indeed a `types.Type` (a `Type` where the kind is a `StructKind`). It's not what CEL expects, it wants a `types.Type` where the kind is a `TypeKind` wrapping the `Type` with the `StructKind`.

**Comment by eddycharly** [comment]
@jpbetz i added a simple unit test, TBH i'm not sure this code is exercised today and that's probably why it doesn't cause issues in practice.

**Review by jpbetz — APPROVED** [review]

**Comment by jpbetz** [comment]
/approve
/lgtm

Thanks for finding and fixing!

**Comment by eddycharly** [comment]
@jpbetz i had to push a fix to please the tests linter

**Pull request merged**